cardazim/server.py

```diff
@@ -54,14 +76,25 @@ def run_listener_server(server_ip: str, server_port: int) -> None:
     :param server_port:
     :type int:
     """
+    print_lock = threading.Lock()
+
     with init_server_socket(server_ip, server_port) as server_socket:
 
         while True:
             client_socket, client_addr = server_socket.accept()
+            
+            threading.Thread(target=listener_thread, kwargs={"print_lock":print_lock, "client_socket":client_socket}).start()
 
-            from_client = read_all_data(client_socket)
-            print (f'Received message: {from_client}')
-
+    
+    # Wait for all threads to finish.
+    # 
+    # this code is unreachable, but if one day I'd want to support quitting while running the server,
+    # I would have then to break from the loop and reach this code.
+    for thread in threading.enumerate():
+        if thread is not threading.main_thread():  # Skip the main thread
+            thread.join()
```

Comment:
if this code is not reachable then don`t write it
no need to do premature code writing